h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -750,109 +766,173 @@ private void readStoreHeader() {
             // the system time was set to the past:
             // we change the creation time
             creationTime = now;
-            storeHeader.put("created", creationTime);
+            storeHeader.put(HDR_CREATED, creationTime);
         }
 
-        // bugfix - data lost issue when partial write occurs at end of file store.
-        // @since 2019-07-31 little-pan
-        newest = readChunkHeaderAndFooterFromStoreTail(newest);
+        long fileSize = fileStore.size();
+        long blocksInStore = fileSize / BLOCK_SIZE;
 
-        long blocksInStore = fileStore.size() / BLOCK_SIZE;
-        // this queue will hold potential candidates for lastChunk to fall back to
-        Queue<Chunk> lastChunkCandidates = new PriorityQueue<>(Math.max(32, (int)(blocksInStore / 4) + 1),
-                new Comparator<Chunk>() {
+        Comparator<Chunk> chunkComparator = new Comparator<Chunk>() {
             @Override
             public int compare(Chunk one, Chunk two) {
                 int result = Long.compare(two.version, one.version);
                 if (result == 0) {
-                    // out of two versions of the same chunk we prefer the one
+                    // out of two copies of the same chunk we prefer the one
                     // close to the beginning of file (presumably later version)
```

Comment:
this comment doesn't make sense (I know you didn't add it) - the newer block would be the one at the __end__ of the file, surely?

Reply:
We can encounter two copies of the same chunk at different locations, if chunk was moved. And newer copy will probably be closer to the beginning of the file (but not necessary, if write was abruptly terminated during two-staged chunk move). So comment is valid IMHO.

Reply:
Ideally, we could add attribute to chunk metadata, something like "revision",  which would be increased on each chunk move. Then we can order on that one instead of file location.